packages/react-components/react-progress/src/components/Progress/useProgressState.tsx

```diff
@@ -0,0 +1,35 @@
+import type { ProgressState, ProgressProps } from './Progress.types';
+import { progressCssVars } from './useProgressStyles';
+
+// if the percentComplete is near 0, don't animate it.
+// This prevents animations on reset to 0 scenarios
+const ZERO_THRESHOLD = 0.01;
+
+export const useProgressState_unstable = (state: ProgressState, props: ProgressProps) => {
```

Comment:
I thought we got rid of useComponentState hooks in favor of useComponent hooks?

Reply:
I thought it would be a bit cleaner to separate it, to keep the useProgress file simple

Reply:
You probably want to put this logic back and follow the same pattern we use elsewhere.